include/transport/time_queue.h

```diff
@@ -360,11 +387,19 @@ namespace qtransport {
                 auto& [bucket_index, value_index, expiry_tick] = _queue.at(_queue_index);
                 auto& bucket = _buckets.at(bucket_index);
 
-                if (value_index >= bucket.size() || ticks > expiry_tick) {
+                if (value_index >= bucket.size() || (ticks > expiry_tick && !in_spike(ticks))) {
+                    std::cerr << "===> front Object has expired"
+                              << " queue_index: " <<_queue_index << " queue_size: " << _queue.size()
+                              << " value_index: " << value_index
+                              << " bucket_index: " << bucket_index
+                              << " bucket_size: " << bucket.size()
+                              << " tick_delta: " << _timer_ctx.delta
+                              << " " << ticks << " > " << expiry_tick
+                              << std::endl;
                     _queue_index++;
                     continue;
                 }
-
+                
```

Comment:
Why are 
```cpp
_queue.clear();
_queue_index = 0;
```
Missing from here? They should be here... for cleanup. Perhaps those missing lines are what necessitated the `move_front` method that was just added?

Reply:
It's the same as it was before, after the while loop.  We only move forward up to the max size of the queue. If at the end, it clears/resets it.  Move will also do it if move results in wrap.